masonry/src/event_loop_runner.rs

```diff
@@ -732,6 +732,17 @@ impl MasonryState<'_> {
                 render_root::RenderRootSignal::ShowWindowMenu(position) => {
                     window.show_window_menu(position);
                 }
+                render_root::RenderRootSignal::WidgetSelectedInInspector(widget_id) => {
+                    let (widget, state) = self.render_root.widget_arena.get_pair(widget_id);
+                    let widget_name = widget.item.short_type_name();
+                    let display_name = if let Some(debug_text) = widget.item.get_debug_text() {
+                        format!("{widget_name}<{debug_text}>")
+                    } else {
+                        widget_name.into()
+                    };
+                    println!("Widget selected in inspector: {widget_id} - {display_name}");
+                    println!("{:#?}", state.item);
+                }
```

Comment:
There would be reasonable arguments for printing the broad shape of the children tree as well. Not necessarily the full widget state, but the names and debug text. Maybe? limited to a certain depth(/length?)